src/libexec/zmrestoreblob

```diff
@@ -138,9 +140,16 @@ if ($orphaned_header_check) {
                     last;
                 }
             }
+            if ( $msg_match_flag == 1 ) {
+                last;
+            }
         }
         if ( $msg_match_flag == 0 ) {
-            print "[] INFO: File $matched_blob not found in backup \n";
+            my @blob_part = split( '-', $org_blob_part );
+            my $msg_id = $blob_part[0];
+            if ( !check_msg_in_shared_blobs( $msg_id, $matched_blob ) ) {
+                print "[] INFO: File $matched_blob not found in backup \n";
```

Comment:
nit: might as well remove that trailing space before the newline since you're in here modifying the code
thought: would it be worth while to list the zip file that contained the blob being restored?  Not sure if that's of interest or not, but figured I'd throw it out there.